code/modules/cm_aliens/hivebuffs/hivebuff.dm

```diff
@@ -0,0 +1,517 @@
+		if(tier > HIVEBUFF_TIER_MINOR)
+			xeno_announcement(engage_flavourmessage, hive.hivenumber, "Buff Purchased")
+		for(var/mob/xenomorph as anything in hive.totalXenos)
+			if(!xenomorph.client)
+				continue
+			xenomorph.play_screen_text(engage_flavourmessage, override_color = "#740064")
+			if(tier <= HIVEBUFF_TIER_MINOR)
+				to_chat(xenomorph, SPAN_XENO(engage_flavourmessage))
+	if(marine_flavourmessage)
+		marine_announcement(marine_flavourmessage, COMMAND_ANNOUNCE, 'sound/AI/bioscan.ogg')
+
+/datum/hivebuff/proc/_announce_buff_cease()
+	for(var/mob/living/xenomorph as anything in hive.totalXenos)
+		if(!xenomorph.client)
+			continue
+		xenomorph.play_screen_text(cease_flavourmessage, override_color = "#740064")
+		to_chat(xenomorph, SPAN_XENO(cease_flavourmessage))
+
+///Signal handler for new xenomorphs joining the hive
+/datum/hivebuff/proc/_handle_xenomorph_new(mob/living/carbon/xenomorph/new_xeno)
+	SIGNAL_HANDLER
+	if(!apply_on_new_xeno)
+		return
+	// If we're the same hive as the buff
+	if(new_xeno.hive == hive)
+		apply_buff_effects(new_xeno)
+
+///The actual effects of the buff to apply
+/datum/hivebuff/proc/apply_buff_effects(mob/living/carbon/xenomorph/xeno)
+	return
+
+/// Reverse the effects here, should be the opposite of apply_effects()
+/datum/hivebuff/proc/remove_buff_effects(mob/living/carbon/xenomorph/xeno)
+	return
+
+/datum/hivebuff/proc/handle_special_checks()
+	return TRUE
+
+////////////////////////////////
+//		BUFFS
+////////////////////////////////
+
+/datum/hivebuff/extra_larva
+	name = "Surge of Larva"
+	desc = "Provides 5 larva instantly to the hive."
+	radial_icon = "larba"
+
+	engage_flavourmessage = "The Queen has purchased 5 extra larva to join the hive!"
+	cost = 5
+	number_of_required_pylons = 1
+	is_reusable = FALSE
+
+/datum/hivebuff/extra_larva/on_engage()
+	hive.stored_larva += 5
+	return TRUE
+
+/datum/hivebuff/extra_life
+	name = "Boon of Plenty"
+	desc = "Increases all xenomorph health by 5% for 10 minutes"
+	tier = HIVEBUFF_TIER_MINOR
+	engage_flavourmessage = "The Queen has imbued us with greater fortitude."
+	duration = 10 MINUTES
+	number_of_required_pylons = 1
+
+/datum/hivebuff/extra_life/apply_buff_effects(mob/living/carbon/xenomorph/xeno)
+	xeno.maxHealth *= 1.05
+
+/datum/hivebuff/extra_life/remove_buff_effects(mob/living/carbon/xenomorph/xeno)
+	xeno.maxHealth = initial(xeno.caste.max_health)
+
+/datum/hivebuff/extra_life/major
+	name = "Major Boon of Plenty"
+	desc = "Increases all xenomorph health by 10% for 10 minutes"
+	tier = HIVEBUFF_TIER_MAJOR
+
+	engage_flavourmessage = "The Queen has imbued us with greater fortitude."
+	duration = 10 MINUTES
+	cost = 2
+	number_of_required_pylons = 2
+	radial_icon = "health_m"
+
+/datum/hivebuff/extra_life/major/apply_buff_effects(mob/living/carbon/xenomorph/xeno)
+	xeno.maxHealth *= 1.1
+
+/datum/hivebuff/extra_life/major/remove_buff_effects(mob/living/carbon/xenomorph/xeno)
+	xeno.maxHealth = initial(xeno.caste.max_health)
```

Comment:
These modifications should be applied to `health_modifier` as + / - because you will lose the maxHealth modification once `recalculate_health` is called.